File: 1º Semestre/Lista 3/Ex2.py
#Escreva um programa para encontrar o valor fatorial de um número n digitado pelo usuário.
#O fatorial de um número n é da forma n! = n * (n -1) * (n - 2) * ... * 2 * 1
num=int(input("Digite um número: "));
n=num-1;
cont=num*n;
res=cont;
while n!=1:
    cont=res;
    n-=1;
    res=cont*n;
print(res);

File: 2º Semestre/Lista 2/Ex2.java
// Fazer um programa para que o usuário entre com um número inteiro. Após isso, o programa deve calcular o termo correspondente da sequência de Fibonacci. Primeiros números da sequência: 0, 1, 1, 2, 3, 5 ...
// Para se calcular o próximo termo da sequência de Fibonacci, é necessário somar os dois termos anteriores.

import java.util.Scanner;

public class Ex2 {
    static long fibonacci(int n) {
        if (n < 2) {
            return n;
        } else {
            return fibonacci(n - 1) + fibonacci(n - 2);
        }
    }
    public static void main(String args []) {
        try (Scanner obj = new Scanner(System.in)) {
            System.out.println("Digite um número: ");
            int num = obj.nextInt();
            System.out.println("A sequência é: ");
            for (int i = 0; i < num; i++) {
                System.out.print(Ex2.fibonacci(i) + "\t");
            }
        }

    }
}